tfjs-layers/src/engine/container.ts

```diff
@@ -652,6 +655,26 @@ export abstract class Container extends Layer {
     batchSetValue(weightValueTuples);
   }
 
+  protected parseWeights(weights: NamedTensorMap) {
+    Object.keys(weights).forEach((key) => {
+      const listParts = key.split('/');
+      const list = ['vars', 'layer_checkpoint_dependencies'];
+      const newKey = listParts
+                         .map(str => {
+                           if (str.startsWith('_')) {
+                             return str.slice(1);
+                           }
+                           return str;
+                         })
+                         .filter(str => !list.includes(str))
+                         .join('/');
```

Comment:
Nit: Can you please add some comments explaining why we do this mapping and filtering? Thanks!

Reply:
Thanks for review! Added some comments, let me know if that make sense to you!